python/ctsm/site_and_regional/tower_site.py

```diff
@@ -0,0 +1,413 @@
+        case.set_value("RUN_REFCASE", os.path.basename(ref_case_root))
+        refdate = None
+        for reffile in glob.iglob(refrundir + "/{}{}.clm2.r.*.nc".format(self.name, root)):
+            m_searched = re.search(r"(\d\d\d\d-\d\d-\d\d)-\d\d\d\d\d.nc", reffile)
+            if m_searched:
+                refdate = m_searched.group(1)
+            symlink_force(reffile, os.path.join(rundir, os.path.basename(reffile)))
+        logger.info("Found refdate of %s", refdate)
+        if not refdate:
+            logger.warning("Could not find refcase for %s", case_root)
+            return False
+
+        for rpfile in glob.iglob(refrundir + "/rpointer*"):
+            safe_copy(rpfile, rundir)
+        if not os.path.isdir(os.path.join(rundir, "inputdata")) and os.path.isdir(
+            os.path.join(refrundir, "inputdata")
+        ):
+            symlink_force(os.path.join(refrundir, "inputdata"), os.path.join(rundir, "inputdata"))
+
+        case.set_value("RUN_REFDATE", refdate)
+        if case_root.endswith(".postad"):
+            case.set_value("RUN_STARTDATE", refdate)
+        # NOTE: if start options are set, RUN_STARTDATE should be modified here
+        return True
+
+    # pylint: disable=too-many-statements
+    def run_case(
+        self,
+        base_case_root,
+        run_type,
+        prism,
+        run_length,
+        user_version,
+        tower_type,
+        user_mods_dirs,
+        overwrite=False,
+        setup_only=False,
+        no_batch=False,
+        rerun=False,
+        experiment=False,
+    ):
+        """
+        Run case.
+
+        Args:
+        self
+        base_case_root: str, opt
+            file path of base case
+        run_type: str, opt
+            transient, post_ad, or ad case, default transient
+        prism: bool, opt
+            if True, use PRISM precipitation, default False
+        run_length: str, opt
+            length of run, default '4Y'
+        user_version: str, opt
+            default 'latest'
+        overwrite: bool, opt
+            default False
+        setup_only: bool, opt
+            default False; if True, set up but do not run case
+        no_batch: bool, opt
+            default False
+        rerun: bool, opt
+            default False
+        experiment: str, opt
+            name of experiment, default False
+        """
+        expect(
+            os.path.isdir(base_case_root),
+            "Error base case does not exist in {}".format(base_case_root),
+        )
+        # -- if user gives a version:
+        if user_version:
+            version = user_version
+        else:
+            version = "latest"
+
+        print("using this version:", version)
+
+        if experiment is not False:
+            self.name = self.name + "." + experiment
+        case_root = os.path.abspath(os.path.join(base_case_root, "..", self.name + "." + run_type))
+
+        rundir = None
+        if os.path.isdir(case_root):
+            if overwrite:
```

Comment:
This section seems long and should be broken out, which might make it testable.

This would be something for later.

Reply:
#2413